### ISSUE TITLE
issue по поводу адресов и также для списка :)

### ISSUE THREAD

**Issue by uliss4ever** [uliss4ever/My_repository_2#6]
Лёш, я там сделала в Дата поправки, в days_counter. Но до конца не уверена, всё ли поняла. Тесты ещё немного добавила, где понимаю. На тестах сеттера застряла)) Или я опять слона не заметила. Выгрузила генератор (там только пока рандом для страны и города убрала, попробовала. И по поводу связного списка. Я послушаю запись и сформулирую, что там должно быть. Но оно правда кажется проще, чем всё предыдущее)
Но поскольку сегодня вмешались дополнительные факторы (как будто того, что было, мне было мало @@@), то я уже не особо стремлюсь и рассчитываю на зачёт в субботу. Главное понимание.
Спасибо тебе!!! 
P.S. коммит вот только что ещё один, самый актуальный)
Да, если декорировать не генератор, то что должен «не запускать» декоратор? Так что точно фабрика для него. Но я сначала обрадовалась - мы же можем чек_дата задекорировать и дальше не пойти. Но если он проверит и найдёт что-то не то, он может же только вывалить ошибку. А дополнительное поведение функции у нас тут может навесить только декоратор.

**Comment by uliss4ever** [comment]
```
"""
Двусвязный список на основе односвязного списка.
    Самостоятельное задание. В двусвязном списке должны быть следующие методы:
    - **`__str__`**
    - **`__repr__`**
    - **`__getitem__`**
    - **`__setitem__`**
    - **`__len__`**
    - **`insert`**
    - **`index`**
    - **`remove`**
    - **`append`**
    - **`__iter__`**
    Необязательно все эти методы должны быть переопределены в явном виде. По максимуму используйте
    наследование, если поведение списков в контексте реализации указанных метод схоже.
    С точки зрения наследования по минимуму перегружайте методы. При необходимости рефакторите базовый класс,
    чтобы локализовать части кода во вспомогательные функции, которые имеют различное поведение
    в связном и двусвязном списках.
    Стремитесь к минимизации кода в дочернем классе.
    Есть какой-то метод класса DoubleLinkedList хотите отработать в явном виде ещё раз, не возбраняется.
"""

# ToDo импорт любой вашей реалиазации LinkedList


class DoubleLinkedList(...):
    class DoubleLinkedNode(...):
        ...
```

**Comment by uliss4ever** [comment]
@Lex-2008 Лёш, я вот про замену не до конца поняла, вообще всё, что node ? 
И то, что .data ? (видимо жа, поскольку красит)
![Снимок экрана 2021-03-03 в 20 10 37](https://user-images.githubusercontent.com/32711430/109844026-c4788400-7c5c-11eb-9250-8195a8ce40f5.png)

![Снимок экрана 2021-03-03 в 20 11 32](https://user-images.githubusercontent.com/32711430/109844014-c2aec080-7c5c-11eb-82bc-610a14d226df.png)

**Comment by Lex-2008** [comment]
Нет, извини - только `Node` с большой буквы надо заменить на `self.node_type`. Это должно встречаться только в двух местах: в функциях `append` и `insert`.

на этих скринах менять ничего не надо - должно быть как на первом скрине :)

**Comment by uliss4ever** [comment]
Я тоже уже подумала, что Node, иначе будет по две точки) Щас)

**Comment by uliss4ever** [comment]
Это именно в классе обычной ноды же?
![Снимок экрана 2021-03-03 в 20 37 16](https://user-images.githubusercontent.com/32711430/109847510-53d36680-7c60-11eb-96bc-cd6a24b4a104.png)

**Comment by Lex-2008** [comment]
Да, только не `tail`, а `node_type` :)

`tail` - это у нас указатель на последний элемент списка, а `node_type` - это тип элементов в этом списке (`Node` или `DoubleNode`)

**Comment by uliss4ever** [comment]
Да что же такое со мной, прости. Я self услышала. И type услышала и дальше пыталась совместить. Но я правда пока про это не до конца понимаю. Новый коммит наверное пока не буду, локально поменяю только?

Короче, он ругается. Если типизация, то она вроде с двоеточием... (это в параметрах). Короче видеть, слышать и иметь это node_type в данных местах он не хочет совсем-совсем...

**Comment by Lex-2008** [comment]
хмм. А можешь сделать коммит в этот репо, я посмотрю?

Может быть, действительно то что я придумал - так нельзя.

**Comment by uliss4ever** [comment]
Да, вот он. https://github.com/uliss4ever/My_repository_2/commit/09be62ba0730db21ec100301b71ca840cb783dff Правда там первое, что я исправила, в двух местах. Ту ноду, что с маленькой буквы. Тебе это понятно будет, что надо обратно или ещё совсем новый коммит лучше?

Пошла посмотрела, а он же показывает, что удаляла - удобно)

**Comment by Lex-2008** [comment]
я поправил - сможешь скачать себе изменения с гитхаба?

**Comment by uliss4ever** [comment]
Пасиб тебе! Я точно могу копипастом, только не знаю куда смотреть. И мёрж он тут вообще не предлагает, я так по-дурацки репо создала видимо

**Comment by Lex-2008** [comment]
Это ты про кнопку в правом верхнем углу пичарма, где синяя стрелка вниз, да?

Он всегда предлагает. Сначала спрашивает, потом смотрит, надо ли было спрашивать. :) Выбери rebase и галку "больше не спрашивать".

**Comment by uliss4ever** [comment]
Не, я про то, где посмотреть твои изменения и как их скачать. Я просто вижу свой коммит 2 часа назад и всё)
Еноты вышел, проверил, а у меня не тот коммит (или пуш забыла...) Но там всё равно и в новом не всё как надо. Сказал мало тестов. Я их пишу. 
Да, ещё про iadd сказал, можно там функцию-замыкание сделать, но это если я хочу, так тоже хорошо

**Comment by Lex-2008** [comment]
Действительно, извини пожалуйста, туплю - у меня пичарм с гитом не-до-подружился и молчит.

Должно быть вот [так](http://home.shpakovsky.ru/Screenshot_2021-03-03_21-26-04.png)

надо в двух местах изменить `self.tail` на `self._node_type`

**Comment by uliss4ever** [comment]
Да что ты, всё в порядке и спасибо!! Когда они подружатся, заживёшь, пичарм лапушка.
Я сегодня шла и думала про эту байку, помнишь? https://yandex.ru/turbo/anekdot.ru/s/id/482226/
По идее он должен был принести 11 батонов. Или тут ТЗ не очень точно сформулировано? Один батон нести в любом случае и взять ещё 10 в случае наличия яиц или либо 1 либо 10... Проблема, короче)

**Comment by Lex-2008** [comment]
> тут ТЗ не очень точно сформулировано

Да, скорее так. По правилам русского языка там вообще подразумевается последнее существительное.

Как в предложении "Вася красил забор в шапке". Кто был в шапке? Последнее стоящее перед шапкой существительное.

Типа такого:
````
что=батон
сколько=1
купить(что, сколько)
что=яйца
if есть(что):
  сколько=десяток
  купить(что, сколько)
````

**Comment by uliss4ever** [comment]
Даааа) Но если уже речь идёт о об условии покупки десяти батонов, то это условие - есть яйца или нет. А один батон - он до if, то есть его надо купить в любом случае) 
Раньше мне это просто шуткой казалось, ну типа программисты живут в своём мире и оторваны от реальности. А сейчас как-то я уже не так категорична...

**Comment by Lex-2008** [comment]
А, да, понял тебя. Неправильный программист просто, потому и анекдот. Он же должен был купить 11 батонов, а взял только 10, ха-ха-ха! :rofl:

**Comment by uliss4ever** [comment]
Вот именно! )))
